src/Microsoft.DotNet.ProjectModel.Loader/LoaderProjectContextExtensions.cs

```diff
@@ -3,47 +3,98 @@
+using System.Linq;
 using System.Reflection;
 using System.Runtime.Loader;
+using Microsoft.DotNet.ProjectModel.Compilation;
+using Microsoft.Extensions.DependencyModel;
 
 namespace Microsoft.DotNet.ProjectModel.Loader
 {
     public static class LoaderProjectContextExtensions
     {
         public static AssemblyLoadContext CreateLoadContext(
-            this ProjectContext context, 
-            string configuration = "Debug", 
-            string outputPath = null)
+            this ProjectContext context,
+            string configuration) => CreateLoadContext(context, context.RuntimeIdentifier, configuration);
+
+        public static AssemblyLoadContext CreateLoadContext(
+            this ProjectContext context,
+            string runtimeIdentifier,
+            string configuration) => CreateLoadContext(context, runtimeIdentifier, configuration);
+
+        public static AssemblyLoadContext CreateLoadContext(
+            this ProjectContext context,
+            string runtimeIdentifier,
+            string configuration,
+            string outputPath)
         {
             var exporter = context.CreateExporter(configuration);
             var assemblies = new Dictionary<AssemblyName, string>(AssemblyNameComparer.OrdinalIgnoreCase);
-            var dllImports = new Dictionary<string, string>();
+            var nativeLibs = new Dictionary<string, string>();
+            var rids = DependencyContext.Default?.RuntimeGraph ?? Enumerable.Empty<RuntimeFallbacks>();
 
             foreach (var export in exporter.GetAllExports())
             {
-                // TODO: Handle resource assemblies
-                foreach (var asset in export.RuntimeAssemblyGroups.GetDefaultAssets())
+                // Process managed assets
+                var group = string.IsNullOrEmpty(runtimeIdentifier) ?
+                    export.RuntimeAssemblyGroups.GetDefaultGroup() :
+                    GetGroup(export.RuntimeAssemblyGroups, runtimeIdentifier, rids.FirstOrDefault(r => r.Runtime.Equals(runtimeIdentifier)));
+                if(group != null)
                 {
-                    // REVIEW: Should we use the following?
-                    // AssemblyLoadContext.GetAssemblyName(asset.ResolvedPath);
-                    var assemblyName = new AssemblyName(asset.Name);
-                    assemblies[assemblyName] = asset.ResolvedPath;
+                    foreach(var asset in group.Assets)
+                    {
+                        assemblies[asset.GetAssemblyName()] = asset.ResolvedPath;
+                    }
                 }
 
-                foreach (var asset in export.NativeLibraryGroups.GetDefaultAssets())
+                // Process native assets
+                group = string.IsNullOrEmpty(runtimeIdentifier) ?
+                    export.NativeLibraryGroups.GetDefaultGroup() :
+                    GetGroup(export.NativeLibraryGroups, runtimeIdentifier, rids.FirstOrDefault(r => r.Runtime.Equals(runtimeIdentifier)));
```

Comment:
Minor perf tweak. You don't need to loop over rids repeatedly. This could be lifted out of the for loop.